components/date-time/README.md

```diff
@@ -58,6 +58,13 @@ The localization for the display of the date and time.
 - Type: `string`
 - Required: No
 
+### format
+
+The format to transform the date and time to on save.
+
+- Type: `string`
+- Required: No
+
```

Comment:
I wonder if we should add this prop. Two issues I have with it:

 - The default WP format (not the format to show, but the format saved and returned by the API) is the `TIMEZONELESS_FORMAT`, that's the reason we're forcing it.

 - This creates a duplication with `is12Hour` but not always. Someone could say we could compute `is12Hour` from the format passed in but that's not always true because we always pass the `TIMEZONELESS_FORMAT` no matter what "display" format we use.

So my opinion is that it's better to keep the format fixed and if you want to display the date in another format, use moment (or something else) outside the component, like we do in the PostSchedule component.